### PR TITLE
[v2] Move os_firewall_allow from defaults to role dependencies.

### PR DESCRIPTION
**Pull request by abutcher**
* Move `os_firewall_allow` from defaults to role dependencies for all roles. These defaults were not being acted on.
* Fix issue with openshift-loadbalancer refactor, ensuring that `haproxy_frontend_port` is fed from the playbook to drive firewall configuration.

The loadbalancer refactor changes were merged post cut so, if we're doing backports for critical fixes, these changes shouldn't be included in the update since the update won't be impacted by them.

https://bugzilla.redhat.com/show_bug.cgi?id=1339431

@detiber PTAL

### CODE REVIEW

**Comment by abutcher** [comment]
@sdodson also PTAL

**Comment by abutcher** [comment]
aos-ci-test

File: roles/openshift_node/meta/main.yml [hunk 1]
```diff
@@ -19,4 +19,17 @@ dependencies:
...
+    port: 80/tcp
+  - service: https
+    port: 443/tcp
+  - service: Openshift kubelet ReadOnlyPort
+    port: 10255/tcp
+  - service: Openshift kubelet ReadOnlyPort udp
+    port: 10255/udp
+  - service: OpenShift OVS sdn
+    port: 4789/udp
+    when: openshift.node.use_openshift_sdn | bool
```
Comment (detiber) [inline roles/openshift_node/meta/main.yml]: hmm, does this when actually work, or just add a 'when' key to the dict?

**Comment by detiber** [comment]
outside of the nitpicky comment about an existing when clause, lgtm.

File: roles/openshift_node/meta/main.yml (hunk 1 above)
Comment (abutcher) [inline roles/openshift_node/meta/main.yml]: It applies -- not sure about the negative case but I can confirm.

Comment (abutcher) [inline roles/openshift_node/meta/main.yml]: I'd assume it just adds this no matter what with a 'when' key in the dict.

Comment (detiber) [inline roles/openshift_node/meta/main.yml]: That's what I was thinking as well...  Unless I added some special hackery to os_firewall, which isn't out of the question either.

Comment (sdodson) [inline roles/openshift_node/meta/main.yml]: Probably doesn't work as expected, though it's not a change from current implementation.

**Comment by sdodson** [comment]
LGTM

**Pull request merged**